### ISSUE TITLE
NPE from updatePlayerStatus function

### ISSUE THREAD

**Issue by DavidEdwards** [googlecast/CastCompanionLibrary-android#287]
In [this code](https://github.com/googlecast/CastCompanionLibrary-android/blob/master/src/com/google/android/libraries/cast/companionlibrary/cast/player/VideoCastControllerFragment.java#L480):

```
case MediaStatus.IDLE_REASON_FINISHED:
    if (!mIsFresh && mMediaStatus.getLoadingItemId()
            == MediaQueueItem.INVALID_ITEM_ID) {
        mCastController.closeActivity();
    }
    break;
```

This variable `mMediaStatus` is (in rare unreproducable circumstances) null. Causing an NPE.

```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.package.name/com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.gms.cast.ah.k()' on a null object reference
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077)
       at android.app.ActivityThread.-wrap15(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5417)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.gms.cast.ah.k()' on a null object reference
       at com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerFragment.updatePlayerStatus(SourceFile:469)
       at com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerFragment.onConfigurationChanged(SourceFile:734)
       at com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerActivity.onCreate(SourceFile:135)
       at android.app.Activity.performCreate(Activity.java:6251)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077)
       at android.app.ActivityThread.-wrap15(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5417)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by naddaf** [comment]
Yes, I should check for a null value there. Expect next update to have a fix for that.

**Comment by naddaf** [comment]
Addressed in v2.8.4
